File: Scala/src/main/java/com/tay/myException/Exception1.scala
package com.tay.myException

/**
 * @author karlieswift 
 *         date: 2020/7/5 14:08 
 *         ClassName: Exception1  
 * @version java "13.0.1"
 *
 *          Scala中的异常不区分所谓的编译时异常和运行时异常，
 *          也无需显示抛出方法异常，所以Scala中没有throws关键字。
 */
object Exception1 {

  def main(args: Array[String]): Unit = {
    try{
      var i=1/0
    }catch {
      case x: ArithmeticException => {
        println("算数异常")
      }
      case y: Exception => {
        println("其他异常")
      }
    }
//    finally {
//      println("finally")
//    }

  }
}

File: java/javase/src/exception_test/Exception1.java
package exception_test;



public class Exception1{ 
	public static String output=""; 
	public static void foo(int i){ 
		try{ 
			if(i==1)
				throw new Exception(); 	
			output+="1"; 
		} 
		catch(Exception e){ 
			output+="2"; 
			return; //如果加上return则原来的134234——>13423,finally一定执行
		} 
		finally{ 
			output+="3"; 
		} 
		output+="4"; 
	}
	public static void main(String args[]){ 
		foo(0);
		System.out.println(output);//134
		foo(1); 
		System.out.println(output);// 134234
	}
}
